kubectl/kubectl.go

```diff
@@ -87,21 +87,85 @@ func (k *Kubectl) GetPodNames(namespace string, selector string) ([]string, erro
 	return namesR, nil
 }
 
+// WaitForNamespaceWithPod blocks until pods matching the selector are available in the specified namespace. It fails after the timeout.
+func (k *Kubectl) WaitForNamespaceWithPod(namespace string, labelName string) error {
+	return wait.PollImmediate(k.pollInterval, k.PollTimeout, func() (bool, error) {
+		return k.NamespaceWithPodExists(namespace, labelName)
+	})
+}
+
+// NamespaceWithPodExists returns true if pods by that label are present in the given namespace
+func (k *Kubectl) NamespaceWithPodExists(namespace string, labelName string) (bool, error) {
+	pods, err := k.GetPodNames(namespace, labelName)
+	if err != nil {
+		return false, err
+	}
+	for _, p := range pods {
+		e, _ := k.PodStatus(namespace, p)
+		if e == nil || e.ContainerStatuses == nil || len(e.ContainerStatuses) == 0 {
+			return false, nil
+		}
+		if !e.ContainerStatuses[0].Ready {
```

Comment:
This assumes there is one container in the pod, maybe we can extend that for each container in the pod?

Reply:
Sure, the last commit should do the job.